lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
"8-bit-ascii" does not help, since its just as platform dependent as saying nothing.

When converting a Binary to a String using '%s', the encoding will be "UTF-8". The most consistent approach would be to then say that 's' here, also means "UTF-8". If not, then converting to and through using '%s' and 's' will give somewhat surprising results.

My preference would therefore be to say that the default encoding is "UTF-8" but that other encodings can be specified. We must then either not allow "8-bit-ascii" since it doesn't exist outside the Ruby community or allow it but raise a red flag stating that it is 100% platform specific and should be avoided at length. My preference would be the former and instead provide a set of encodings known to be standardized.